pkg/parquet/reader.go

```diff
@@ -220,6 +220,17 @@ func (r *s3Reader) ReadFile(ctx context.Context, file string) (ReadResults, erro
 	}
 
 	size := int(pr.GetNumRows())
+
+	if batchSize != nil {
+		err := pr.SkipRows(offset)
+
+		if err != nil {
+			return nil, err
+		}
+
+		size = mdl.EmptyIntIfNil(batchSize)
```

Comment:
imagine we have a parquet file of 99 entries and a batch size of 10, would we need to ensure that size=9 for the last batch?